crates/flox-rust-sdk/src/flox.rs

```diff
@@ -551,6 +552,42 @@ impl Flox {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct GitHubUser {
+    login: String,
+}
+
+pub struct GitHubClient {
+    base_url: String,
+    oauth_token: String,
+}
+
+impl GitHubClient {
+    pub async fn new(base_url: &str, oauth_token: &str) -> Self {
+        GitHubClient {
+            base_url: base_url.to_string(),
+            oauth_token: oauth_token.to_string(),
+        }
+    }
+
+    pub async fn get_username(&self) -> Result<String, reqwest::Error> {
+        let url = format!("{}/user", self.base_url);
+        let client = reqwest::Client::new();
+        let response = client
+            .get(&url)
+            .header("Authorization", format!("token {}", self.oauth_token))
+            .send()
+            .await?;
+
+        if response.status().is_success() {
+            let user: GitHubUser = response.json().await?;
+            Ok(user.login)
+        } else {
+            Err(response.error_for_status().unwrap_err())
```

Comment:
nit: I think this unwrap could cause a panic if a 304 is returned, if I'm looking at the right docs https://docs.github.com/en/free-pro-team@latest/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user--status-codes